### PR TITLE
New comment generic element

### PR DESCRIPTION
**Pull request by atm-arnaud**
# New comment generic element
This new pull request change many thing about my comment system. We want comment system working with all Dolibarr object (commonObject) So we managed it to works.

We have one question. If we want, we could include (action and bloc tpl part) into note tpl. With this change, we could include commentary system in every note page on dolibarr. But we will lost admin conf part to allow or not commentary system. 

What do you think about it ? We continue like this (1 tab for comment) and implement it for all object we need or we include it in note tab? (I prefer a new tab but it needs more works)

### CODE REVIEW

**Comment by atm-arnaud** [comment]
@eldy what do you think about ? This feature works. It's just to improve my last NEW comment system on task. Now it could be used everywhere.

**Comment by eldy** [comment]
There is already a system to add comment, that works everywhere, on object.
It is the "event" (we can have an event "comment added").All events are visible on bottom right of objects.
With v7, they are sorted by date decrease so we get the last event. And on the tab "Events" we can see all events including "comment addition".
So this feature seems to be a duplicate of existing feature. What is difference ?

**Comment by atm-arnaud** [comment]
There is a difference between the first sense of event and comment. As I think, we can't regroup the commentary system into events because it's already a group of calendar event and appointments.

About comments, we need a different view. I don't know if you try my PR to see the template design. It's like a SMS feature. Direct comment under others with personalize background user color. 

comments-project

At my last argument, I'm looking to github. Imagine a Pull Request with Event system. Discussion will be hard to follow with all clicks to back in list and see answer.  It can't be clear and easy to use. We will add too much complicity to the main feature. I'm just thinking about users case. 

May could we contact us on line to continue this discussion. (It's hard for me to explain in English)

**Comment by eldy** [comment]
The difference is only in how information is reported on screen. I don't see so much differences (because my vision is in the future).
The curent event feature will surely evolve to offer a way to show event in a nicer look like you suggest. But it is just a way to show a list
The form to post a "quick" event will probably appear too on traditional (current) event feature.
A CRM must provide a 360 degree view  (including comment, but also emails, business events, etc...),
For this reason, i will keep this feature as "hidden" because the future event feature will be very close to this new one.

**Comment by hregis** [comment]
@atm-arnaud @eldy 
you delete this table but this table is used again in the code !!

`DROP TABLE llx_projet_task_comment;`

**Comment by hregis** [comment]
@atm-arnaud @eldy 
i disable this in my PR + fix phpcs : https://github.com/Dolibarr/dolibarr/pull/7520

**Comment by BebZ** [comment]
True that comment and event are different, should have at least different styles of display if it is automatic, or user inputed.
Notes are really practical on almost all objects, but there is also "event" that is more automatic, and a user doen't think about create an event to put a note. It should be cleared / cleaned.

There is also a plethoric offer of theses external dlistore modules, it should be good to think about taking all the pros of these modules and put it in the Core (history module have a lot of usefulness when changind things on propals orders and invoices with user-tracked modifications)

BTW history module kills the "comments" tab i guess
